### PR TITLE
feat: add transformOperations

### PR DESCRIPTION
**Pull request by marbemac**
Turns open api file into array of http operations.

### CODE REVIEW

File: src/oas/computeOperations.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
+import { IHttpOperation } from '@stoplight/types';
+import { flatten, get, keys, map } from 'lodash';
+import { OpenAPIObject } from 'openapi3-ts';
+import { Spec } from 'swagger-schema-official';
+
+import { HttpOperationTransformer } from '../oas/types';
+
+const methods = ['get', 'post', 'put', 'delete', 'options', 'head', 'patch', 'trace'];
+
+export function computeOasOperations(
```
Comment (pytlesk4) [inline src/oas/computeOperations.ts]: Nice, I like that it doesn't care that it's a oas2/3 document. Can we do something similar for the `transformOas2Operation` and `transformOas3Operation` and make a `trasnformOasOperation`?

Comment (pytlesk4) [reply src/oas/computeOperations.ts]: Is it required to pass in the transformers?

Can we just check the openapi or swagger property?

File: src/oas/computeOperations.ts [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  document: Spec | OpenAPIObject,
+  transformer: HttpOperationTransformer<any>,
```
Comment (P0lip) [inline src/oas/computeOperations.ts]:
```suggestion
  transformer: HttpOperationTransformer<unknown>,
```

**Comment by pytlesk4** [comment]
@marbemac I need this function, do you have time to finish this out?

**Comment by XVincentX** [comment]
> I suspect it all comes down to the fact we don't trust the input, but if we are unsure whether a proper document is provided, we could simply specify unknown and use type guards instead.

I agree with this so much. We're really, *really* indisciplined with the usage of `any` and this brought more than 1 problem in Prism, for example.

I have an idea how to solve this problem *FOREVER*, but you'd probably kill me for that :trollface:

**Pull request merged**